sql/range.go

```diff
@@ -21,15 +21,35 @@ import (
 )
 
 // RangeCollection is a collection of ranges that represent different (non-overlapping) filter expressions.
-type RangeCollection []Range
+type RangeCollection interface {
+	Equals(otherCollection RangeCollection) (bool, error)
+	DebugString() string
+	String() string
+	ToRanges() []Range
+}
+
+// MySQLRangeCollection is a collection of MySQL ranges that represent different (non-overlapping) filter expressions.
+type MySQLRangeCollection []MySQLRange
+
+// Range represents a set iteration over an integrator's index. Ranges are not required to be contiguous, although they
+// are expected to not overlap.
+type Range interface {
+	Equals(other Range) (bool, error)
+	String() string
+	DebugString() string
+}
 
-// Range is a collection of RangeColumns that are ordered by the column expressions as returned by their parent
+// MySQLRange is a collection of RangeColumns that are ordered by the column expressions as returned by their parent
 // index. A single range represents a set of values intended for iteration by an integrator's index.
-type Range []RangeColumnExpr
+type MySQLRange []MySQLRangeColumnExpr
```

Comment:
Probably want these types in at least another file, and maybe another package (`sql/mysql` or similar)